File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/message/MessageDao.java
package com.github.dedis.popstellar.repository.database.message;

import androidx.room.*;

import com.github.dedis.popstellar.model.objects.security.MessageID;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface MessageDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(MessageEntity message);

  @Query("SELECT * FROM messages WHERE message_id = :messageId")
  MessageEntity getMessageById(MessageID messageId);

  @Query("SELECT * FROM messages LIMIT :n")
  Single<List<MessageEntity>> takeFirstNMessages(int n);
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/message/MessageDao.kt
package com.github.dedis.popstellar.repository.database.message

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.dedis.popstellar.model.objects.security.MessageID
import io.reactivex.Completable
import io.reactivex.Single

@Dao
interface MessageDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE) fun insert(message: MessageEntity): Completable

  @Query("SELECT * FROM messages WHERE message_id = :messageId")
  fun getMessageById(messageId: MessageID): MessageEntity?

  @Query("SELECT * FROM messages LIMIT :n")
  fun takeFirstNMessages(n: Int): Single<List<MessageEntity>?>
}
